workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-artic-ivar-analysis/pe-wgs-ivar-analysis.ga

```diff
@@ -0,0 +1,1215 @@
+            "name": "Text transformation",
+            "outputs": [
+                {
+                    "name": "output",
+                    "type": "input"
+                }
+            ],
+            "position": {
+                "bottom": 114.93333435058594,
+                "height": 113.60000610351562,
+                "left": 989.7666625976562,
+                "right": 1189.7666625976562,
+                "top": 1.3333282470703125,
+                "width": 200,
+                "x": 989.7666625976562,
+                "y": 1.3333282470703125
+            },
+            "post_job_actions": {
+                "HideDatasetActionoutput": {
+                    "action_arguments": {},
+                    "action_type": "HideDatasetAction",
+                    "output_name": "output"
+                }
+            },
+            "tool_id": "toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1",
+            "tool_shed_repository": {
+                "changeset_revision": "fb4ff3c42cd3",
+                "name": "text_processing",
+                "owner": "bgruening",
+                "tool_shed": "toolshed.g2.bx.psu.edu"
+            },
+            "tool_state": "{\"adv_opts\": {\"adv_opts_selector\": \"basic\", \"__current_case__\": 0}, \"code\": \"/^[^#]/s/^[^\\\\t]+\\\\t/NC_045512.2\\\\t/\", \"infile\": {\"__class__\": \"ConnectedValue\"}, \"__page__\": null, \"__rerun_remap_job_id__\": null}",
+            "tool_version": "1.1.1",
+            "type": "tool",
+            "uuid": "564fa633-a537-4d7e-8f30-ebba672fce68",
+            "workflow_outputs": []
+        },
+        "15": {
+            "annotation": "",
+            "content_id": "toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1",
+            "errors": null,
+            "id": 15,
+            "input_connections": {
+                "infile": {
+                    "id": 12,
+                    "output_name": "consensus"
+                }
+            },
+            "inputs": [
+                {
+                    "description": "runtime parameter for tool Text transformation",
+                    "name": "infile"
+                }
+            ],
+            "label": "Consensus genome (masked for depth)",
+            "name": "Text transformation",
+            "outputs": [
+                {
+                    "name": "output",
+                    "type": "input"
+                }
+            ],
+            "position": {
+                "bottom": 481.33331298828125,
+                "height": 134,
+                "left": 752.75,
+                "right": 952.75,
+                "top": 347.33331298828125,
+                "width": 200,
+                "x": 752.75,
+                "y": 347.33331298828125
+            },
+            "post_job_actions": {
+                "RenameDatasetActionoutput": {
+                    "action_arguments": {
+                        "newname": "#{infile}"
```

Comment:
Can you find a better name here?

What this does is it will create datasets named "ivar consensus on data xxx Consensus" just like the previous step (i.e. the name will be misleading both in terms of the tool and the input dataset #). I know you're hiding the previous step's output collection, but if you ever need to debug one of the resulting histories, you'll regret this choice.

Reply:
In my experience, this renames the outputs to have the name of the collection element - i.e. if the input is called `sample1` the output is `sample1` in the collection. The naming of the collection as a whole remains `"iv.ar consensus on data xxx Consensus`.

Reply:
The renaming applies to the collection and its datasets. It's the collection elements' element identifers that are passed through like you say. In other words when you're viewing the datasets in the context of the collection you will see the names you're expecting, but when you view them as individual datasets (e.g. when you unhide hidden elements) or try to edit their names, you will see the "newname" set here.
Anyway, that newname is misleading for the collection and its elements.